ApplicationCode/GeoMech/GeoMechDataModel/RigFemPartResultCalculatorMudWeightWindow.cpp

```diff
@@ -0,0 +1,323 @@
+
+#include "RigFemPartResultCalculatorMudWeightWindow.h"
+
+#include "RigFemPart.h"
+#include "RigFemPartCollection.h"
+#include "RigFemPartGrid.h"
+#include "RigFemPartResultsCollection.h"
+#include "RigFemResultAddress.h"
+#include "RigFemScalarResultFrames.h"
+#include "RigGeoMechBoreHoleStressCalculator.h"
+#include "RigGeoMechWellLogExtractor.h"
+
+#include "RiaOffshoreSphericalCoords.h"
+
+#include "cafProgressInfo.h"
+
+#include "cvfBoundingBox.h"
+
+#include <QString>
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RigFemPartResultCalculatorMudWeightWindow::RigFemPartResultCalculatorMudWeightWindow( RigFemPartResultsCollection& collection )
+    : RigFemPartResultCalculator( collection )
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RigFemPartResultCalculatorMudWeightWindow::~RigFemPartResultCalculatorMudWeightWindow()
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RigFemPartResultCalculatorMudWeightWindow::isMatching( const RigFemResultAddress& resVarAddr ) const
+{
+    return ( resVarAddr.fieldName == "MUD-WEIGHT" &&
+             ( resVarAddr.componentName == "MWW" || resVarAddr.componentName == "MWM" ||
+               resVarAddr.componentName == "UMWL" || resVarAddr.componentName == "LMWL" ) );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RigFemScalarResultFrames* RigFemPartResultCalculatorMudWeightWindow::calculate( int                        partIndex,
+                                                                                const RigFemResultAddress& resVarAddr )
+{
+    CVF_ASSERT( isMatching( resVarAddr ) );
+
+    caf::ProgressInfo frameCountProgress( m_resultCollection->frameCount() * 4, "" );
+    frameCountProgress.setProgressDescription( "Calculating Mud Weight Window" );
+
+    double wellPathDeviation = 12.3;
+    double wellPathAzimuth   = 32.4;
+    double ucsBar            = 3.4;
+    double poissonsRatio     = 0.45;
+    double K0_FG             = 0.445;
+    double airGap            = 123.0;
+
+    UpperLimitParameter upperLimitParameter = UpperLimitParameter::FG;
+    LowerLimitParameter lowerLimitParameter = LowerLimitParameter::MAX_OF_PORE_PRESSURE_AND_SFG;
+
+    // Pore pressure
+    frameCountProgress.setNextProgressIncrement( m_resultCollection->frameCount() );
```

Comment:
The progress increments would be better served using ProgressInfo::task

See: RicNewFractureModelPlotFeature::createPlot( RimEclipseCase* eclipseCase, RimFractureModel* fractureModel, int timeStep )